test/integration/serverTest.js

```diff
@@ -154,6 +157,150 @@ describe('Brave Sensor server', () => {
     await redis.disconnect()
   })
 
+  describe('POST request: alerts from firmware state machine for locations with a non-null Particle Siren Id', () => {
+    beforeEach(async () => {
+      await redis.clearKeys()
+      await db.clearSessions()
+      await db.clearLocations()
+      await db.clearClients()
+      const client = await clientFactory(db)
+      await db.createLocation(
+        testLocation1Id,
+        testLocation1PhoneNumber,
+        MOVEMENT_THRESHOLD,
+        STILLNESS_TIMER,
+        DURATION_TIMER,
+        1000,
+        INITIAL_TIMER,
+        ['+15005550006'],
+        '+15005550006',
+        ['+15005550006'],
+        1000,
+        'locationName',
+        door_coreID,
+        radar_coreID,
+        'XeThru',
+        'alertApiKey',
+        true,
+        true,
+        'particleCoreIdTest',
+        client.id,
+      )
+      sandbox.stub(braveAlerter, 'startAlertSession')
+      sandbox.stub(braveAlerter, 'sendSingleAlert')
+      sandbox.spy(particle, 'callFunction')
+      sandbox.spy(helpers, 'logError')
+    })
+    afterEach(async () => {
+      await redis.clearKeys()
+      await db.clearSessions()
+      await db.clearLocations()
+      await db.clearClients()
+      sandbox.restore()
+      helpers.log('\n')
+    })
+
+    it('should return 200 to a request with no headers', async () => {
+      const response = await chai.request(server).post('/api/sensorEvent').send({})
+      expect(response).to.have.status(200)
+    })
+
+    it('should log an error for a request with no headers', async () => {
+      await chai.request(server).post('/api/sensorEvent').send({})
+      expect(helpers.logError).to.have.been.called
+    })
+
+    it('should return 200 for a valid request', async () => {
+      const response = await firmwareAlert(radar_coreID, SENSOR_EVENT.STILLNESS)
+      expect(response).to.have.status(200)
+    })
+
+    it('should call particle.callFunction with the correct device ID if sirenParticleId is not null', async () => {
+      await firmwareAlert(radar_coreID, SENSOR_EVENT.STILLNESS)
+      expect(particle.callFunction).to.have.been.calledWith({
+        deviceId: 'particleCoreIdTest',
+        name: 'start',
+        argument: 'start',
+        auth: helpers.getEnvVar('PARTICLE_ACCESS_TOKEN'),
+      })
+    })
+
+    it('should not call startAlertSession if particleSirenID is not null', async () => {
+      const sessions = await db.getAllSessionsFromLocation(testLocation1Id)
+      expect(sessions.length).to.equal(0)
```

Comment:
The name of this test doesn't quite correspond with what you're actually testing. 
I'd either change the name of the test to `'should not create a new session if particleSirenID is not null`
or change the test to spy on the `braveAlerter.startAlertSession` function and use `expect(braveAlerter.startAlertSession).not.to.be.called` (my syntax might be off, but something like that).

Reply:
make sure to also add a call to firmwareAlert as the test stimulus for the API - right now its just running this db function and expect on an empty database that's had no input to it.

Reply:
Changed the name of the test and added a call to firmwareAlert